app/controllers/partners/PartnerSummaryController.scala

```diff
@@ -0,0 +1,98 @@
+import featureswitch.core.config.TaskList
+import forms.partners.PartnerSummaryForm
+import models.Entity
+import play.api.mvc.{Action, AnyContent}
+import services.AttachmentsService.AdditionalPartnersDocumentsAnswer
+import services.{AttachmentsService, EntityService, SessionProfile, SessionService}
+import uk.gov.hmrc.auth.core.AuthConnector
+import viewmodels.PartnerSummaryRow
+import views.html.partners.PartnerSummary
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class PartnerSummaryController @Inject()(val authConnector: AuthConnector,
+                                         val sessionService: SessionService,
+                                         entityService: EntityService,
+                                         attachmentsService: AttachmentsService,
+                                         view: PartnerSummary)
+                                        (implicit appConfig: FrontendAppConfig,
+                                     val executionContext: ExecutionContext,
+                                     baseControllerComponents: BaseControllerComponents)
+  extends BaseController with SessionProfile {
+
+  def show: Action[AnyContent] = isAuthenticatedWithProfile {
+    implicit request =>
+      implicit profile =>
+        entityService.getAllEntities(profile.registrationId).map {
+          case Nil => Redirect(controllers.routes.TaskListController.show)
+          case partners if partners.size == 1 => Redirect(routes.PartnerEntityTypeController.showPartnerType(minPartnerIndex))
+          case partners => Ok(view(PartnerSummaryForm(), buildRows(partners), partners.size))
+        }
+  }
+
+
+  def submit: Action[AnyContent] = isAuthenticatedWithProfile {
+    implicit request =>
+      implicit profile =>
+        PartnerSummaryForm().bindFromRequest.fold(
+          errors =>
+            entityService.getAllEntities(profile.registrationId).map {
+              case Nil => Redirect(controllers.routes.TaskListController.show)
+              case partnerEntities => BadRequest(view(errors, buildRows(partnerEntities), partnerEntities.size))
+            },
+          addMore =>
+            if (addMore) {
+              entityService.getNextValidIndex.flatMap { nextIndex =>
+                if (nextIndex > appConfig.maxPartnerCount) {
+                  attachmentsService.storeAttachmentDetails(profile.registrationId, AdditionalPartnersDocumentsAnswer(true)).map(_ => {
+                    Redirect(routes.AdditionalPartnerEntityController.show)
+                  })
+                } else {
+                  Future.successful(Redirect(routes.PartnerEntityTypeController.showPartnerType(nextIndex)))
+                }
+              }
+            } else {
+              attachmentsService.storeAttachmentDetails(profile.registrationId, AdditionalPartnersDocumentsAnswer(false)).map(_ => {
+                if (isEnabled(TaskList)) {
```

Comment:
tasklist FS check not really necessary here as this partner flow was built with the assumption it is always on